src/core/agent/services/signifyNotificationService.test.ts

```diff
@@ -669,4 +693,194 @@ describe("Signify notification service of agent", () => {
       ConnectionHistoryType.CREDENTIAL_REQUEST_AGREE
     );
   });
+
+  test("Should handle long operations", async () => {
```

Comment:
Hmm, can we create a new describe section called "long running operation tracker" and put these these new tests in there. And in that describe, have a beforeAll that spys on setTimeout once.

Once we have that in place, we can have a separate tests for each type of long running operation rather than one big test for all of the positive test cases which is a bit harder to follow